lib/dpl/provider/chef_supermarket.rb

```diff
@@ -20,8 +21,16 @@ def check_auth
         error "#{options[:client_key]} does not exist" unless ::File.exist?(options[:client_key])
       end
 
+      def cookbook_metadata
+        # credit: https://coderwall.com/p/wwt0sw/easiest-way-to-parse-a-cookbook-s-metadata-file
+        metadata_file = 'metadata.rb'
+        # read in metadata
+        @metadata = Chef::Cookbook::Metadata.new
+        @metadata.from_file(metadata_file)
```

Comment:
If `metadata.rb` is the only file that can be used (for one reason or another), it does not make sense to use an intermediary.

Simply:

```ruby
def cookbook_name
  return unless File.readable?('metadata.rb')
  @cookbook_name ||= Chef::Cookbook::Metadata.new('metadata.rb').name
end
```

would suffice (And use `#cookbook_name` to defined `@cookbook_name` in `#check_app`.)